content/articles/data-preprocessing-python/index.md

```diff
@@ -0,0 +1,248 @@
+
+![Verification_code](/engineering-education/data-preprocessing-python/Cpo.png)
+
+We then copy and paste this code in the authorization code box on Colab interface and click **Ctrl + Enter**. This shows that our drive is mounted.
+
+Now that our drive it's successfully mounted in Google Colab, we proceed and import the libraries we shall require to work with along our journey.
+To import these libraries, let's type and run the code below.
+
+
+### Step 1: Importing the Libraries
+
+```Python
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+```
+Next, we proceed and import the dataset.
+
+### Step 2: Import the dataset
+
+Let's type and run the following code.
+
+```python
+Dataset = pd.read_csv("/content/drive/MyDrive/Dataset.csv")
+
+# importing an array of features
+x = Dataset.iloc[:, :-1].values 
+# importing an array of dependent variable
+y = Dataset.iloc[:, -1].values
+```
+
+Above, we specified two variables, x for the features and y for the dependent variable. The features set, as declared in the code `Dataset.iloc[:, :-1]` will consist of all rows and columns of our dataset except the last column. Similarly, the dependent variable y will consist of all rows but only the last column as declared in the code `Dataset.iloc[:, -1].values`
+
+Let's have a look at our data by executing the code.
+
+```Python
+print(x) # returns an array of features
+```
+
+![Features_Array](/engineering-education/data-preprocessing-python/Feature.array.png)
+
+```Python
+print(y) # viewing an array of the dependent variable.
+```
+![Features_Array](/engineering-education/data-preprocessing-python/dependent_variable.png)
+
+Our data is imported successfully in the form of an array of features x and a dependent variable y.
+
+### Step 3: Taking care of the missing data
+
+We notice that x, an array of our features, contains missing values in the Age and Salary column from our output. We need to deal with this problem before we can implement our machine learning models on our data. There exist two techniques that we use to deal with the missing value:
+
+1. **Deleting the observation reporting the missing value(s):**
+
+This technique works well with big datasets with few missing values. For instance, deleting a row from a dataset with hundreds of observations cannot affect the information quality of the dataset. However, this technique is not suitable for the dataset reporting many missing values. Deleting many rows from a dataset can lead to information loss information.
+
+To ensure no crucial information is lost, we need to make use of an appropriate technique.
+
+2. **Replace the missing value(s) with the average of the variable in which the data is missing:**
+
+This technique has reported no harm to the dataset's quality; therefore, it's the technique that we most use to handle missing values.
```

Comment:
How are you basing this statement? If you claim that filling missing value with averages does no harm to the dataset, provide justification. This is an important statement.